### PR TITLE
fix: Delete feature external resources when GitHub integration was deleted

### PR DESCRIPTION
**Pull request by novakzaballa**
…deleted

Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

- Delete feature external resources when GitHub  integration was deleted 

## How did you test this code?

- Unit test

### CODE REVIEW

File: api/integrations/github/models.py [hunk 1]
```diff
@@ -47,3 +48,19 @@ class Meta:
                 name="unique_repository_data",
             )
         ]
+
+    @hook(BEFORE_DELETE)
+    def delete_feature_external_respurces(
```
Comment (khvn26) [inline api/integrations/github/models.py]:
```suggestion
    def delete_feature_external_resources(
```

File: api/integrations/github/models.py [hunk 2]
```diff
@@ -47,3 +48,19 @@ class Meta:
...
+        self,
+    ) -> None:
+        from features.feature_external_resources.models import (
+            FeatureExternalResource,
+        )
+
+        FeatureExternalResource.objects.filter(
+            feature__in=self.project.features.all(),
+            type__in=[
```
Comment (khvn26) [inline api/integrations/github/models.py:62]: Do you think we need to add an index on `FeatureExternalResource.type`?

File: api/tests/unit/integrations/github/test_unit_github_views.py [hunk 3]
```diff
@@ -191,18 +192,28 @@ def test_cannot_create_github_repository_due_to_unique_constraint(
 def test_github_delete_repository(
     client: APIClient,
     organisation: Organisation,
+    feature_external_resource: FeatureExternalResource,
     github_configuration: GithubConfiguration,
     github_repository: GithubRepository,
+    mocker,
```
Comment (khvn26) [inline api/tests/unit/integrations/github/test_unit_github_views.py]: `from pytest_mock import MockerFixture`

```suggestion
    mocker: MockerFixture,
```

File: api/integrations/github/models.py [hunk 4]
```diff
@@ -47,3 +48,19 @@ class Meta:
...
+    @hook(BEFORE_DELETE)
+    def delete_feature_external_respurces(
+        self,
+    ) -> None:
+        from features.feature_external_resources.models import (
+            FeatureExternalResource,
+        )
+
+        FeatureExternalResource.objects.filter(
+            feature__in=self.project.features.all(),
```
Comment (khvn26) [inline api/integrations/github/models.py]: I'd avoid a `SELECT *` here:

```suggestion
            feature_id__in=self.project.features.values_list("id", flat=True),
```

**Review by khvn26 — CHANGES_REQUESTED** [review]

File: api/integrations/github/models.py (hunk 1 above)
Comment (novakzaballa) [reply api/integrations/github/models.py]: Good catch, was corrected

File: api/integrations/github/models.py (hunk 2 above)
Comment (novakzaballa) [reply api/integrations/github/models.py:62]: Good idea, I added an index

File: api/tests/unit/integrations/github/test_unit_github_views.py (hunk 3 above)
Comment (novakzaballa) [reply api/tests/unit/integrations/github/test_unit_github_views.py]: Added

File: api/integrations/github/models.py (hunk 4 above)
Comment (novakzaballa) [reply api/integrations/github/models.py]: Done.

**Review by khvn26 — APPROVED** [review]

**Pull request merged**